src/main/java/teammates/storage/api/AccountsDb.java

```diff
@@ -184,72 +182,67 @@ public void updateAccount(AccountAttributes a)
 
         for (AccountAttributes accountToDelete : accounts) {
-            if (!accountToDelete.studentProfile.pictureKey.isEmpty()) {
-                deletePicture(new BlobKey(accountToDelete.studentProfile.pictureKey));
-            }
+            deleteAccount(accountToDelete.googleId);
         }
-        deleteEntities(accounts);
-        closePm();
     }
 
     private Account getAccountEntity(String googleId, boolean retrieveStudentProfile) {
-
-        try {
-            Key key = KeyFactory.createKey(Account.class.getSimpleName(), googleId);
-            Account account = getPm().getObjectById(Account.class, key);
-
-            if (JDOHelper.isDeleted(account)) {
-                return null;
-            } else if (retrieveStudentProfile && account.getStudentProfile() == null) {
-                // This situation cannot be reproduced and hence not tested
-                // This only happens when existing data in the store do not have a profile
-                account.setStudentProfile(new StudentProfile(account.getGoogleId()));
-            }
-
-            return account;
-        } catch (IllegalArgumentException iae) {
-            return null;
-        } catch (JDOObjectNotFoundException je) {
+        Account account = load().id(googleId).now();
+        if (account == null) {
             return null;
         }
+
+        account.setIsStudentProfileEnabled(retrieveStudentProfile);
+
+        return account;
     }
 
     private Account getAccountEntity(String googleId) {
         return getAccountEntity(googleId, false);
     }
 
     @Override
-    protected Object getEntity(EntityAttributes entity) {
-        return getAccountEntity(((AccountAttributes) entity).googleId);
+    protected LoadType<Account> load() {
+        return ofy().load().type(Account.class);
     }
 
     @Override
-    protected QueryWithParams getEntityKeyOnlyQuery(EntityAttributes attributes) {
-        Class<?> entityClass = Account.class;
-        String primaryKeyName = Account.PRIMARY_KEY_NAME;
-        AccountAttributes aa = (AccountAttributes) attributes;
-        String id = aa.googleId;
+    protected Account getEntity(AccountAttributes entity) {
+        return getAccountEntity(entity.googleId);
+    }
 
-        Query q = getPm().newQuery(entityClass);
-        q.declareParameters("String idParam");
-        q.setFilter(primaryKeyName + " == idParam");
+    @Override
+    protected QueryKeys<Account> getEntityQueryKeys(AccountAttributes attributes) {
+        Key<Account> keyToFind = Key.create(Account.class, attributes.googleId);
+        return load().filterKey(keyToFind).keys();
+    }
+
+    @Override
+    protected AccountAttributes makeAttributes(Account entity) {
+        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, entity);
 
-        return new QueryWithParams(q, new Object[] {id}, primaryKeyName);
+        return new AccountAttributes(entity);
```

Comment:
Yup it was designed that way so only the constructor varies across the different implementations, while the variable name `entity` remains the same.